pkg/utils/controller/controller.go

```diff
@@ -51,7 +52,7 @@ var (
 // NewUnexpectedBehaviorError returns ErrUnexpectedBehavior type error when err is not nil.
 func NewUnexpectedBehaviorError(err error) error {
 	if err != nil {
-		klog.ErrorS(err, "Unexpected behavior identified by the controller")
+		klog.ErrorS(err, "Unexpected behavior identified by the controller", "stackTrace", debug.Stack())
```

Comment:
can we paste an example of how the error message look in the PR?

Reply:
can we allow the caller to determine if they want to print the stack?

Reply:
here is an example,

E0704 07:07:42.025370       1 controller/controller.go:55] "Unexpected behavior identified by the controller" err="ResourceGroupHashAnnotation is not set" stackTrace="goroutine 1281 [running]:\nruntime/debug.Stack()\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x5e\ngo.goms.io/fleet/pkg/utils/controller.NewUnexpectedBehaviorError({0x1cdd720, 0xc000e77190})\n\t/workspace/pkg/utils/controller/controller.go:55 +0x48\ngo.goms.io/fleet/pkg/controllers/clusterresourceplacement.(*Reconciler).getOrCreateClusterResourceSnapshot(0xc0002e65b0, {0x1cf7da8, 0xc001209810}, 0xc001750380, 0x0, 0xc002103680, 0xa)\n\t/workspace/pkg/controllers/clusterresourceplacement/controller.go:433 +0x3c6\ngo.goms.io/fleet/pkg/controllers/clusterresourceplacement.(*Reconciler).handleUpdate(0xc0002e65b0, {0x1cf7da8, 0xc001209810}, 0xc001750380)\n\t/workspace/pkg/controllers/clusterresourceplacement/controller.go:188 +0x5bb\ngo.goms.io/fleet/pkg/controllers/clusterresourceplacement.(*Reconciler).Reconcile(0xc0002e65b0, {0x1cf7da8, 0xc001209810}, {0x1771ea0?, 0xc001330c40?})\n\t/workspace/pkg/controllers/clusterresourceplacement/controller.go:78 +0x75a\ngo.goms.io/fleet/pkg/utils/controller.(*controller).reconcileHandler(0xc000019620, {0x1cf7da8, 0xc001209810}, {0x1771ea0, 0xc001330c40})\n\t/workspace/pkg/utils/controller/controller.go:253 +0xf0\ngo.goms.io/fleet/pkg/utils/controller.(*controller).processNextWorkItem(0xc000019620, {0x1cf7da8, 0xc001209810})\n\t/workspace/pkg/utils/controller/controller.go:240 +0x1be\ngo.goms.io/fleet/pkg/utils/controller.(*controller).Run.func2()\n\t/workspace/pkg/utils/controller/controller.go:213 +0x89\ncreated by go.goms.io/fleet/pkg/utils/controller.(*controller).Run in goroutine 795\n\t/workspace/pkg/utils/controller/controller.go:208 +0x225\n"

We can improve our alerts to show the error and stacktrace in a separate column.

Reply:
> can we allow the caller to determine if they want to print the stack? Sometimes this can flood the log for more frequent cases like HB not recieved.

In this case, would like to create a separate warning instead of using unexpectedError. If the member is stopped, it's an expected error.

Let the upper level decide whether to trigger the alert or not, cause 1) easy to adjust the threshold 2) need more context (e.g. member cluster provisioning state).